File: routes/channel.py
from flask import Blueprint, request, jsonify
import database
from AuthKeyGen import requiresToken
import time
from routes.server import userInServer

bpChannel = Blueprint("channel",__name__)

@bpChannel.route("/createChannel",methods=["POST"])
@requiresToken
def createChannel(user):
    """
  write smth here

    """
    data = request.get_json()
    if not(userInServer(user["userID"],data["serverID"])):
        response = jsonify({
            "Error":"You are not in that serer"
        })

    createdChannel = {
        "name": data["name"],
        "serverID": data["serverID"],
    }
    
    database.addRowAndReturnRowID(createdChannel,"channel")

    response = jsonify({
            "Message":"Channel Created Successfully"
        })
    return response

# get servers a user is in, i admit this is not a great endpoint name but i dont want a roblox looking method name thats 80 words long
@bpChannel.route("/getChannels",methods=["POST"])
@requiresToken
def getChannels(user):

    data = request.get_json()
    if not(userInServer(user["userID"],data["serverID"])):
        response = jsonify({
            "Error":"You are not in that serer"
        })

    channels = database.queryTableValue(["name","id"],"channel","serverID",data["serverID"],True)
    if(not channels):
        response = jsonify({
            "Message":"No channels found"
        })

    channelList = []

    for channel in channels.values():
        channelList.append({"name":channel[0],"id":channel[1]})

    response = jsonify({
            "Message":channelList
        })
    return response

File: routes/channel.go
package routes

import (
	db "RelayServer/database"
	"encoding/json"
	"fmt"
	"net/http"
)

type CreateChannelRequest struct {
	Name     string `json:"name"`
	ServerID string `json:"serverID"`
}

type GetChannelsRequest struct {
	ServerID string `json:"serverID"`
}

func CreateChannel(w http.ResponseWriter, r *http.Request) {
	var data CreateChannelRequest
	err := json.NewDecoder(r.Body).Decode(&data)
	if err != nil {
		fmt.Println(err)
		http.Error(w, "Invalid Credentials", http.StatusUnauthorized)
		return
	}
	user, err := db.AuthHeaderValidation(r)
	if err != nil {
		//do something
	}

	var rawUserID interface{} = user["userID"]
	userID := fmt.Sprintf("%v", rawUserID) //cooked

	if !UserInServer(data.ServerID, userID) {
		//FAH
	}

	createdChannel := map[string]any{
		"name":     data.Name,
		"serverID": data.ServerID,
	}

	_, e := db.AddRowWithIDReturn(createdChannel, "channel")
	if e != nil {

	}
	//do some thing abt check the headers for the JWT and then create serverUser
	fmt.Fprintf(w, "created server")
}

func GetChannels(w http.ResponseWriter, r *http.Request) {
	var data GetChannelsRequest
	err := json.NewDecoder(r.Body).Decode(&data)
	if err != nil {
		http.Error(w, "Invalid Credentials", http.StatusUnauthorized)
		return
	}
	user, err := db.AuthHeaderValidation(r)
	if err != nil {
		//do something
	}

	var rawUserID interface{} = user["userID"]
	userID := fmt.Sprintf("%v", rawUserID) //cooked

	if !UserInServer(data.ServerID, userID) {
		//FAH
	}

	// channels, err := db.Query([]string{"name", "id"}, "channel", "serverID", data.ServerID)
	// channels.
	// fmt.Fprintf(w,channels)
	//this part do something with it

}
